### DISCUSSION TITLE
article/2sat/

### DISCUSSION THREAD

**Discussion by giscus** [3-24/3-24.github.io#24] [Comments]
# logic/2SAT/

2-CNF is CNF that every clause has at most two literals. The satisfiability of CNF is NP-complete in general, however 2-CNF satisfiability is special becuase it is solvable in polynomial time. Let’s call the satisfiability problem of 2-CNF as 2-SAT.

https://3-24.github.io/logic/2SAT/

**Comment by leejseo** [comment]
We can solve 2-QBF in poly-time too :)
https://www.acmicpc.net/problem/16667

**Reply by YoungseokCh** [reply]
Nice!
